### ISSUE TITLE
Callable.rpc() on RPC function throws an error "Invalid call. Nonexistent function 'rpc' in base 'Callable'.

### ISSUE THREAD

**Issue by SpazElectro** [godotengine/godot#87841]
### Tested versions

- Reproducible in: Godot v4.2.1 stable win64

### System information

Windows 10 - Godot v4.2.1.stable

### Issue description

Creating a `Callable` and calling the `rpc` function throws an error stating that the call is invalid, but calling the `rpc` function on the function and not the `Callable` works.

### Steps to reproduce

Create a new GDScript in a blank project, attach it to a node, add the following to the script:
```gdscript
@rpc
func hello():
	print("RPC fn has been called!")

func _ready():
	hello.rpc() # OK
	var fn = Callable(self, "hello")
	fn.rpc() # ERR
```

### Minimal reproduction project (MRP)

[bugreport1.zip](https://github.com/godotengine/godot/files/14131316/bugreport1.zip)

**Comment by AThousandShips** [comment]
This is a limitation of the system, it needs to be an actual RPC callable internally, a named callable like this isn't the same

This might be somewhat difficult to solve, as it would only work in some cases, but I'll look into some of the possible mechanics of it

**Comment by SpazElectro** [comment]
Fixed by #87858!

**Comment by AThousandShips** [comment]
This isn't solved yet, wait for it being merged :)
